packages/shared/src/loadClerkJsScript.ts

```diff
@@ -1,28 +1,52 @@
-import { parsePublishableKey } from '@clerk/shared/keys';
-import { loadScript } from '@clerk/shared/loadScript';
-import { isValidProxyUrl, proxyUrlToAbsoluteURL } from '@clerk/shared/proxy';
-import { addClerkPrefix } from '@clerk/shared/url';
-
-import { errorThrower } from '../errors/errorThrower';
-import type { IsomorphicClerkOptions } from '../types';
-import { isDevOrStagingUrl } from './isDevOrStageUrl';
+import type { ClerkOptions, SDKMetadata, Without } from '@clerk/types';
+
+import { buildErrorThrower } from './error';
+import { createDevOrStagingUrlCache, parsePublishableKey } from './keys';
+import { loadScript } from './loadScript';
+import { isValidProxyUrl, proxyUrlToAbsoluteURL } from './proxy';
+import { addClerkPrefix } from './url';
 import { versionSelector } from './versionSelector';
 
 const FAILED_TO_LOAD_ERROR = 'Clerk: Failed to load Clerk';
 
-type LoadClerkJsScriptOptions = Omit<IsomorphicClerkOptions, 'proxyUrl' | 'domain'> & {
+const { isDevOrStagingUrl } = createDevOrStagingUrlCache();
+
+const errorThrower = buildErrorThrower({ packageName: '@clerk/shared' });
+
+/**
+ * Sets the package name for error messages during ClerkJS script loading.
+ *
+ * @example
+ * setClerkJsLoadingErrorPackage('@clerk/clerk-react');
+ */
+export function setClerkJsLoadingErrorPackageName(packageName: string) {
+  errorThrower.setPackageName({ packageName });
+}
```

Comment:
This one is only used inside `loadClerkJsScript`. I named it like that and not export another `setErrorThrowerOptions` to avoid confusion. We might need to also add this to [frontend SDK spec](https://clerk.com/docs/pr/1368/references/sdk/frontend-only#hotload-clerk-js) @LekoArts 

I also thought of sharing this `errorThrower` across SDKs but that would mess with existing error throwers in next and react.